### ISSUE TITLE
Support AmazonSES v2 API in the SES adapter.

### ISSUE THREAD

**Issue by jadengis** [swoosh/swoosh#847]
**Is your feature request related to a problem? Please describe.**

SES has a v2 API available that has some new features. In particular, SES has some subscription management features that are only available in the v2 API. [Here is the AWS page on subscription management](https://docs.aws.amazon.com/ses/latest/dg/sending-email-subscription-management.html). With the new stricter requirements for senders being imposed by Google and Yahoo, it's going to be pretty important for SES users to access these subscription management features.

**Is your feature request related to a specific adapter? Please specify.**

This is related to the Amazon SES adapter

**Describe the solution you'd like**

Either reimplement the SES adapter using the new API or provide a new adapter that is using the new API and provides access to the new list management options.

**Describe alternatives you've considered**

There isn't really an alternative that's not moving away from using swoosh, or building a new adapter myself.

**Additional context**

[New bulk sender requirements from Google / Yahoo](https://aws.amazon.com/blogs/messaging-and-targeting/an-overview-of-bulk-sender-changes-at-yahoo-gmail/).

**Comment by adamu** [comment]
> There isn't really an alternative that's not moving away from using swoosh,

An alternative would be to handle subscriptions at the application layer, rather than delegating it to SES.

**Comment by princemaple** [comment]
That is true. If it's not very relevant to email sending, it might be a better idea to simply use an AWS API client to manage it. (I haven't really looked into what this feature does)

**Comment by jadengis** [comment]
> An alternative would be to handle subscriptions at the application layer, rather than delegating it to SES.

@adamu I don't really really agree that is an alternative. That is an alternative to using the SES suppression feature, not an alternative way to use these suppression features. In particular though, the problem I'm calling out here is that the SES adapter is currently using an old API that does not have access to new SES features via provider options. List management is just an example of one such feature that I am interested in using.

> That is true. If it's not very relevant to email sending, it might be a better idea to simply use an AWS API client to manage it. (I haven't really looked into what this feature does)

@princemaple This is relevant to email sending, because these list management options are to be included in the actual `SendEmail` API call.

It is very similar to existing provider options that already exist for the AmazonSES adapter e.g. `configuration_set_name`. I imagine that if new SES features are added in the future, they will be added to the v2 API and not the v1 API (which the adapter uses currently).

Ignoring the suppression management features specifically, is there any reason why we wouldn't wany to upgrade the SES adapter to the new version of the `SendEmail` API?

**Comment by princemaple** [comment]
Not against upgrading. Happy to accept pr that upgrades the current one if no breaking change, or a separate one that utilise V2 API.

We don't want to handle anything more than sending emails. That's all.
